crates/core/src/lib.rs

```diff
@@ -97,21 +97,21 @@ impl Default for Config {
         // knobs for each of these settings just yet and instead they're
         // generally set to defaults. Environment-variable-based fallbacks are
         // supported though as an escape valve for if this is a problem.
-        //
-        // NB: much of this will change in Wasmtime 13 as the settings are
-        // different. Ping @alexcrichton for assistance in updating this if
-        // needed (and delete this comment after the 13 update).
         let mut pooling_config = PoolingAllocationConfig::default();
         pooling_config
-            .instance_count(env("SPIN_WASMTIME_INSTANCE_COUNT", 1_000))
-            .instance_size(env("SPIN_WASMTIME_INSTANCE_SIZE", (10 * MB) as u32) as usize)
-            .instance_tables(env("SPIN_WASMTIME_INSTANCE_TABLES", 2))
-            .instance_table_elements(env("SPIN_WASMTIME_INSTANCE_TABLE_ELEMENTS", 30_000))
-            .instance_memories(env("SPIN_WASMTIME_INSTANCE_MEMORIES", 1))
+            .total_component_instances(env("SPIN_WASMTIME_INSTANCE_COUNT", 1_000))
+            .max_component_instance_size(
+                env("SPIN_WASMTIME_INSTANCE_SIZE", (10 * MB) as u32) as usize
+            )
+            .max_tables_per_component(env("SPIN_WASMTIME_INSTANCE_TABLES", 2))
+            .table_elements(env("SPIN_WASMTIME_INSTANCE_TABLE_ELEMENTS", 30_000))
+            .max_memories_per_component(env("SPIN_WASMTIME_INSTANCE_MEMORIES", 1))
```

Comment:
While this will allow instantiating components this would notably disallow instantiating composed components so the default here may want to be higher than 1. This limit is for the entire component loaded into Wasmtime not each individual component itself within that.

(same for tables above)

Reply:
Set this to 10, but I'm not really sure what a good default would be here...

Reply:
@alexcrichton does this limit make each component use the specified number of memory slots in the pooling allocator, or do we only allot the number actually required? If the former, that seems pretty unfortunate, because I think we'll see components with larger compositions pretty soon.

Assuming it's the latter, I think we should bump this up, maybe to 32 or so, because otherwise we'll soon run into issues with at least demos where we want to have more than 10 components in a composition

Reply:
The latter (after Nick's recent refactoring). So yes bumping it to 32 should be fine, that just limits the max concurrent instances if everything happens to use 32

Reply:
Set to 32 and added a comment.